{"name": ["shgo_slsqp_sobol_cube", "shgo_powell_sobol_cube", "shgo_nelder_sobol_cube", "scipy_slsqp_cube", "scipy_powell_cube", "scipy_lbfgsb_cube", "pysot_ei_cube", "pysot_lcb_cube", "pysot_random_cube", "pysot_srbf_cube", "pysot_dycors_cube", "ax_default_cube", "optuna_cmaes_cube", "optuna_tpe_cube", "optuna_random_cube", "optuna_random_cube_clone", "optuna_random_cube_clone_1", "optuna_random_cube_clone_2", "platypus_genetic_cube", "platypus_evolutionary_cube", "platypus_nsgaii_cube", "platypus_cmaes_cube", "platypus_gde3_cube", "platypus_ibea_cube", "platypus_moead_cube", "platypus_omopso_cube", "platypus_smpso_cube", "platypus_spea2_cube", "platypus_epsmoea_cube", "nevergrad_ngopt_cube", "nevergrad_ngopt4_cube", "nevergrad_de_cube", "nevergrad_portfolio_cube", "nevergrad_oneplus_cube", "nevergrad_hammersley_cube", "nevergrad_ngopt8_cube", "hyperopt_tpe_cube", "hyperopt_atpe_cube", "hyperopt_rand_cube", "pymoo_nelder_cube", "pymoo_nsga3_cube", "pymoo_unsga3_cube", "pymoo_pattern_cube", "pymoo_nsga2_cube", "pymoo_unsga3_cube", "skopt_gp_default_cube", "bayesopt_ucb_cube", "bayesopt_ei_cube", "bayesopt_poi_cube", "nlopt_direct_cube", "nlopt_ags_cube", "nlopt_esch_cube", "nlopt_isres_cube", "nlopt_crs2lm_cube", "nlopt_directr_cube", "nlopt_directo_cube", "nlopt_directl_cube", "bobyqa_default_cube", "bobyqa_noise_cube", "ultraopt_random_cube", "ultraopt_forest_cube", "ultraopt_gbrt_cube", "ultraopt_etpe_cube", "dlib_cube", "dlib_default_cube", "hebo_cube_sequential", "hebo_cube_batch", "hebo_sequential_cube", "hebo_batch_cube"], "count": [77, 61, 75, 140, 37, 29, 204, 217, 219, 236, 182, 203, 198, 211, 194, 224, 215, 233, 209, 219, 202, 234, 214, 193, 231, 201, 206, 215, 234, 214, 218, 232, 215, 195, 211, 215, 197, 203, 220, 200, 227, 458, 216, 203, 0, 203, 212, 227, 206, 195, 211, 224, 204, 196, 191, 200, 222, 220, 230, 75, 55, 58, 62, 56, 61, 37, 48, 0, 14], "rating": [1295.8148321912363, 1305.1418871567353, 1228.557360179262, 1696.089435031101, 1296.7151211153082, 1546.2233636202748, 2130.9670038380573, 2029.92117495473, 1068.1006229224045, 2524.3217694861855, 2459.9269084971124, 2393.9023174414374, 1977.434198287909, 2072.99043479783, 1600, 1600, 1600, 1600, 980.9147431109924, 978.2620583135072, 1022.4795970948014, 850.434679508319, 938.7546157707768, 1080.707413723988, 1432.6044232070756, 1538.3362182929925, 1250.2441771410506, 756.8761812975314, 1343.452272926365, 1925.9927094309621, 1969.1100929470076, 1524.8474275607266, 1296.546041793896, 2393.423111862362, 781.0429426835115, 1779.087967537972, 1582.9734663009526, 1635.966894963666, 1579.0828146411054, 2054.620295988391, 2123.629060484343, 2025.1612423903273, 2200.168158244488, 1000.6599739303733, 1600, 2314.4032996670335, 2357.190288844902, 2369.3895305796755, 2375.2928007413593, 2425.2351554891925, 2124.154105288033, 1530.0823355679288, 1077.651536852618, 1356.0824098439343, 2425.659212983906, 1497.3480878879966, 2617.352632020144, 2298.7319385660185, 2385.0727268818264, 1321.6409531181187, 1923.9836627374343, 1839.6625822592334, 1944.237902976556, 1926.4895268001449, 1994.2620226877514, 1925.1161229648428, 1856.4903955959294, 1600, 1815.4267263738975], "traceback": ["passing", "passing", "passing", "passing", "Traceback (most recent call last):\n  File \"/opt/hostedtoolcache/Python/3.8.11/x64/lib/python3.8/site-packages/humpday/comparison/eloratings.py\", line 76, in optimizer_game\n    black_best_val, black_best_x, black_feval_count = black(objective, n_trials=n_black_trials,\n  File \"/opt/hostedtoolcache/Python/3.8.11/x64/lib/python3.8/site-packages/humpday/optimizers/scipycube.py\", line 40, in scipy_powell_cube\n    return scipy_cube(objective=objective, n_trials=n_trials, n_dim=n_dim, with_count=with_count, method='powell')\n  File \"/opt/hostedtoolcache/Python/3.8.11/x64/lib/python3.8/site-packages/humpday/optimizers/scipycube.py\", line 29, in scipy_cube\n    result = minimize(_objective, x0=[0]*n_dim, method=options['method'],bounds=bounds, options=options)\n  File \"/opt/hostedtoolcache/Python/3.8.11/x64/lib/python3.8/site-packages/scipy/optimize/_minimize.py\", line 614, in minimize\n    return _minimize_powell(fun, x0, args, callback, bounds, **options)\n  File \"/opt/hostedtoolcache/Python/3.8.11/x64/lib/python3.8/site-packages/scipy/optimize/optimize.py\", line 3059, in _minimize_powell\n    fx2 = squeeze(func(x2))\n  File \"/opt/hostedtoolcache/Python/3.8.11/x64/lib/python3.8/site-packages/scipy/optimize/optimize.py\", line 464, in function_wrapper\n    return function(np.copy(x), *(wrapper_args + args))\n  File \"/opt/hostedtoolcache/Python/3.8.11/x64/lib/python3.8/site-packages/humpday/optimizers/scipycube.py\", line 27, in _objective\n    return objective(list(x))\n  File \"/opt/hostedtoolcache/Python/3.8.11/x64/lib/python3.8/site-packages/humpday/objectives/portfolio.py\", line 146, in markowitz_realized_on_cube\n    return realized_something_factory(g=g,u=u)\n  File \"/opt/hostedtoolcache/Python/3.8.11/x64/lib/python3.8/site-packages/humpday/objectives/portfolio.py\", line 121, in realized_something_factory\n    w = cube_to_weights(u)\n  File \"/opt/hostedtoolcache/Python/3.8.11/x64/lib/python3.8/site-packages/humpday/objectives/portfolio.py\", line 108, in cube_to_weights\n    raise ValueError(\"u should be in hypercube\")\nValueError: u should be in hypercube\n", "passing", "passing", "Black took 211 function evals when instructed to use 211", "passing", "passing", "passing", "passing", "passing", "passing", "passing", "passing", "passing", "passing", "passing", "passing", "passing", "passing", "passing", "passing", "passing", "passing", "passing", "passing", "passing", "passing", "passing", "passing", "passing", "passing", "passing", "passing", "passing", "passing", "passing", "passing", "passing", "passing", "passing", "passing", "not yet run", "passing", "passing", "passing", "passing", "passing", "passing", "passing", "Black took 211 function evals when instructed to use 211", "passing", "passing", "passing", "passing", "Black took 219 function evals when instructed to use 219", "passing", "passing", "passing", "passing", "passing", "passing", "passing", "passing", "passing", "Traceback (most recent call last):\n  File \"/opt/hostedtoolcache/Python/3.8.10/x64/lib/python3.8/site-packages/hebo/acq_optimizers/evolution_optimizer.py\", line 119, in optimize\n    res   = minimize(prob, algo, ('n_gen', self.iter), verbose = self.verbose)\n  File \"/opt/hostedtoolcache/Python/3.8.10/x64/lib/python3.8/site-packages/pymoo/optimize.py\", line 85, in minimize\n    res = algorithm.solve()\n  File \"/opt/hostedtoolcache/Python/3.8.10/x64/lib/python3.8/site-packages/pymoo/model/algorithm.py\", line 226, in solve\n    self._solve(self.problem)\n  File \"/opt/hostedtoolcache/Python/3.8.10/x64/lib/python3.8/site-packages/pymoo/model/algorithm.py\", line 321, in _solve\n    self.next()\n  File \"/opt/hostedtoolcache/Python/3.8.10/x64/lib/python3.8/site-packages/pymoo/model/algorithm.py\", line 246, in next\n    self._next()\n  File \"/opt/hostedtoolcache/Python/3.8.10/x64/lib/python3.8/site-packages/pymoo/algorithms/genetic_algorithm.py\", line 93, in _next\n    self.off = self.mating.do(self.problem, self.pop, self.n_offsprings, algorithm=self)\n  File \"/opt/hostedtoolcache/Python/3.8.10/x64/lib/python3.8/site-packages/pymoo/model/infill.py\", line 40, in do\n    _off = self.eliminate_duplicates.do(_off, pop, off)\n  File \"/opt/hostedtoolcache/Python/3.8.10/x64/lib/python3.8/site-packages/pymoo/model/duplicate.py\", line 26, in do\n    pop = pop[~self._do(pop, None, np.full(len(pop), False))]\n  File \"/opt/hostedtoolcache/Python/3.8.10/x64/lib/python3.8/site-packages/pymoo/model/duplicate.py\", line 75, in _do\n    D = self.calc_dist(pop, other)\n  File \"/opt/hostedtoolcache/Python/3.8.10/x64/lib/python3.8/site-packages/pymoo/model/duplicate.py\", line 66, in calc_dist\n    D = cdist(X, X)\n  File \"/opt/hostedtoolcache/Python/3.8.10/x64/lib/python3.8/site-packages/pymoo/util/misc.py\", line 90, in cdist\n    return scipy.spatial.distance.cdist(A, B, **kwargs)\n  File \"/opt/hostedtoolcache/Python/3.8.10/x64/lib/python3.8/site-packages/scipy/spatial/distance.py\", line 2954, in cdist\n    return cdist_fn(XA, XB, out=out, **kwargs)\nValueError: Unsupported dtype object\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/opt/hostedtoolcache/Python/3.8.10/x64/lib/python3.8/site-packages/humpday/comparison/eloratings.py\", line 76, in optimizer_game\n    black_best_val, black_best_x, black_feval_count = black(objective, n_trials=n_black_trials,\n  File \"/opt/hostedtoolcache/Python/3.8.10/x64/lib/python3.8/site-packages/humpday/optimizers/hebocube.py\", line 52, in hebo_sequential_cube\n    return hebo_cube_factory(objective=objective, n_trials=n_trials, n_dim=n_dim, with_count=with_count, n_suggestions=1)\n  File \"/opt/hostedtoolcache/Python/3.8.10/x64/lib/python3.8/site-packages/humpday/optimizers/hebocube.py\", line 39, in hebo_cube_factory\n    rec = opt.suggest(n_suggestions=n_suggestions) # <-- don't change this\n  File \"/opt/hostedtoolcache/Python/3.8.10/x64/lib/python3.8/site-packages/hebo/optimizers/hebo.py\", line 124, in suggest\n    rec = opt.optimize(initial_suggest = best_x, fix_input = fix_input).drop_duplicates()\n  File \"/opt/hostedtoolcache/Python/3.8.10/x64/lib/python3.8/site-packages/hebo/acq_optimizers/evolution_optimizer.py\", line 125, in optimize\n    res   = minimize(prob, algo, ('n_gen', self.iter), verbose = self.verbose)\n  File \"/opt/hostedtoolcache/Python/3.8.10/x64/lib/python3.8/site-packages/pymoo/optimize.py\", line 85, in minimize\n    res = algorithm.solve()\n  File \"/opt/hostedtoolcache/Python/3.8.10/x64/lib/python3.8/site-packages/pymoo/model/algorithm.py\", line 226, in solve\n    self._solve(self.problem)\n  File \"/opt/hostedtoolcache/Python/3.8.10/x64/lib/python3.8/site-packages/pymoo/model/algorithm.py\", line 321, in _solve\n    self.next()\n  File \"/opt/hostedtoolcache/Python/3.8.10/x64/lib/python3.8/site-packages/pymoo/model/algorithm.py\", line 246, in next\n    self._next()\n  File \"/opt/hostedtoolcache/Python/3.8.10/x64/lib/python3.8/site-packages/pymoo/algorithms/genetic_algorithm.py\", line 93, in _next\n    self.off = self.mating.do(self.problem, self.pop, self.n_offsprings, algorithm=self)\n  File \"/opt/hostedtoolcache/Python/3.8.10/x64/lib/python3.8/site-packages/pymoo/model/infill.py\", line 40, in do\n    _off = self.eliminate_duplicates.do(_off, pop, off)\n  File \"/opt/hostedtoolcache/Python/3.8.10/x64/lib/python3.8/site-packages/pymoo/model/duplicate.py\", line 26, in do\n    pop = pop[~self._do(pop, None, np.full(len(pop), False))]\n  File \"/opt/hostedtoolcache/Python/3.8.10/x64/lib/python3.8/site-packages/pymoo/model/duplicate.py\", line 75, in _do\n    D = self.calc_dist(pop, other)\n  File \"/opt/hostedtoolcache/Python/3.8.10/x64/lib/python3.8/site-packages/pymoo/model/duplicate.py\", line 66, in calc_dist\n    D = cdist(X, X)\n  File \"/opt/hostedtoolcache/Python/3.8.10/x64/lib/python3.8/site-packages/pymoo/util/misc.py\", line 90, in cdist\n    return scipy.spatial.distance.cdist(A, B, **kwargs)\n  File \"/opt/hostedtoolcache/Python/3.8.10/x64/lib/python3.8/site-packages/scipy/spatial/distance.py\", line 2954, in cdist\n    return cdist_fn(XA, XB, out=out, **kwargs)\nValueError: Unsupported dtype object\n", "passing"], "active": [true, true, true, true, true, true, true, true, true, true, true, false, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true, false, false, false, false]}